crates/astria-composer/src/grpc.rs

```diff
@@ -68,9 +72,10 @@ impl GrpcServer {
             .await;
 
         grpc_server
-            .serve_with_incoming(tokio_stream::wrappers::TcpListenerStream::new(
-                self.listener,
-            ))
+            .serve_with_incoming_shutdown(
+                tokio_stream::wrappers::TcpListenerStream::new(self.listener),
+                self.shutdown_token.cancelled(),
```

Comment:
Noted above: I think this doesn't work.

We need instead some kind of `shutting_down: AtomicBool` that is checked in the server's `ComposerService`, refusing new transactions while still serving health requests.

Reply:
for a follow-up